tasks/kernel_matrix_testing/README.md

```diff
@@ -1,425 +1,564 @@
+-   `--stack=<name>`: The target stack. Defaults to a name based on the current git branch.
+-   `--init-stack`: Automatically creates the stack if it doesn't exist.
+-   `--new`: Creates a fresh configuration file, removing any existing VMs from the stack's configuration.
+-   `--from-ci-pipeline=<id>`: Configures the stack to replicate failed jobs from a CI pipeline. See [CI Integration](#ci-integration).
 
-Configuring the stack involves generating a configuration file which specifies the VMs to launch. Henceforth, this file will be referred to as the `vmsets` file.
+**Specifying VMs (`--vms`)**
 
-The `vmsets` file contains the list of one or more sets of VMs to launch. A set of VMs refers to a collection of VMs sharing some characteristic. The following is an exhaustive list of possible sets:
+The `--vms` list is a comma-separated list of VM entries. Each entry is parsed in a fuzzy manner and must contain three elements separated by dashes (`-`):
 
-- Custom x86_64 kernels on remote x86_64 machine.
-- Custom arm64 kernels on remote arm64 machine.
-- Custom kernels on local machine with corresponding architecture.
-- x86_64 distribution on remote x86_64 machine.
-- arm64 distribution on remote arm64 machine.
-- local distribution with architecutre corresponding to the local machine.
+1.  **Recipe**: `distro` for distribution images or `custom` for custom kernels.
+2.  **Architecture**: `x86_64` (or `amd64`), `arm64` (or `aarch64`), or `local` for your machine's architecture.
+3.  **Version**: The distribution name/version (e.g., `ubuntu22`, `jammy`, `debian10`) or a custom kernel version (e.g., `5.4`).
 
-Sample VMSet file can be found [here](https://github.com/DataDog/test-infra-definitions/blob/f85e7eb2f003b6f9693c851549fbb7f3969b8ade/scenarios/aws/microVMs/sample-vm-config.json).
+The order of elements does not matter.
 
-The file can be generated for a particular `stack` with the `gen-config` task.
-This task takes as parameters
+**Examples:**
 
-- The stack to target specified with [--stack=<name>]
-- The list of VMS to generate specified by --vms=<list>. See [VMs list](#vms-list) for details on how to specify the list.
-- Optional paramenter `--init-stack` to automatically initialize the stack. This can be specified to automatically run the creating stack step.
-- Optional parameter `--new` to generate a fresh configuration file.
+-   `jammy-local-distro`, `distro-local-jammy`, `local-ubuntu22-distro` all resolve to a **local Ubuntu 22.04 distribution VM**.
+-   `amazon4.14-x86-distro`, `distro-x86_64-amazon4.14` all resolve to a **remote x86_64 Amazon Linux 2 (kernel 4.14) distribution VM**.
+-   `custom-arm-5.4`, `5.4-arm64-custom` all resolve to a **remote arm64 custom kernel 5.4 VM**.
 
-The file can be incrementally generated. This means a user may generate a vmset file. Launch it. Add more VMs. Launch them in the same stack.
+#### Incremental Configuration Examples
```

Comment:
This feature does not work correctly as of today.

Reply:
We would like to support this feature at some point, so it is not removed from the code.